src/Ring.cc

```diff
@@ -79,12 +79,44 @@ std::pair<double, int> Ring::computeOptimalRingParametersRectangle(double module
 }
 
 
-void Ring::buildModules(EndcapModule* templ, int numMods, double smallDelta) {
+void Ring::buildModules(EndcapModule* templ, int numMods, double smallDelta, double phiShift, double modTranslateX) {
```

Comment:
This is a bit longer than it used to be. Maybe a few more comments, or moving part of the code in another member function might help clarify?

Reply:
I've added a few more comments, hopefully that helps. Moving part of the code to a new member function is tricky without passing around objects that probably shouldn't be - the only other option is to keep the original buildModules and have a new one that is only called in case of an irregular set of modules (that need the extra functionality). I can do that if you prefer.